### PR TITLE
stateful deployments: add corrections to API structs and methods

### PR DESCRIPTION
**Pull request by pkazmierczak**
This changeset includes changes accidentally left out from [24641](https://github.com/hashicorp/nomad/pull/24641).

### CODE REVIEW

File: api/tasks.go [hunk 1]
```diff
@@ -474,6 +475,7 @@ type VolumeMount struct {
 	Volume          *string `hcl:"volume,optional"`
 	Destination     *string `hcl:"destination,optional"`
 	ReadOnly        *bool   `mapstructure:"read_only" hcl:"read_only,optional"`
+	Sticky          *bool   `mapstructure:"sticky" hcl:"sticky,optional"`
```
Comment (tgross) [inline api/tasks.go:478]: I missed this on the original PR, but the `volume_mount` block is at the task level, not at the allocation level where scheduling is controlled. I don't think it has any meaning here. We should remove it from the `structs` package version as well.

Comment (pkazmierczak) [reply api/tasks.go:478]: oof that's a slip on my part, I fix

Comment (pkazmierczak) [reply api/tasks.go]: fixed in ac2bf73

**Review by tgross — APPROVED** [review]
LGTM :+1:

**Pull request merged**